torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,271 @@
+        """Retrieve image/mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of image, mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        hits = self.index.intersection(tuple(query), objects=True)
+        img_filepaths = cast(List[str], [hit.object for hit in hits])
+        mask_filepaths = [path.replace(path.split("/")[-1], path.split("/")[-2]+"_mask2019.TIF") for path in img_filepaths]
+        
+        # my testing
+        # path = "/Users/yc/Downloads/austral/p74_r92/L71074092_09220011103_B10.TIF"
+        # dir_name, img_name = path.split("/")[-2:]  # ex: dir_name = p74_r92 and img_name = L71074092_09220011103_B10.TIF
+        # mask_path = path.replace(img, dir_name+"_mask2019.TIF")
+        # mask = "/Users/yc/Downloads/austral/p74_r92/p74_r92_mask2019.TIF"
+        
+        # "/Users/yc/Downloads/austral/p228_r97/L72228097_09720011126_B70.TIF"
+        # "/Users/yc/Downloads/austral/p228_r97/p228_r97_mask2019.TIF"
+        
+        # "/Users/yc/Downloads/austral/p231_r93_3/L71231093_09320010811_B10.TIF"
+        # "/Users/yc/Downloads/austral/p231_r93_3/p231_r93_3_mask2019.TIF"
+
+
+        if not img_filepaths:
+            raise IndexError(
+                f"query: {query} not found in index with bounds: {self.bounds}"
+            )
+
+        img = self._merge_files(img_filepaths, query, self.band_indexes)
+        mask = self._merge_files(mask_filepaths, query, self.band_indexes)
+        # Mask needs to be converted from 0, 64, 128, 192, 255 to 0, 1, 2, 3, 4
+        
+        sample = {
+            "crs": self.crs,
+            "bbox": query,
+            "image": img.float(),
+            "mask": mask.long(),
+        }
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def plot(
+        self,
+        sample: Dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        image = np.rollaxis(sample["image"].numpy().astype("uint8").squeeze(), 0, 3)
+        mask = sample["mask"].numpy().astype("uint8").squeeze()
+
+        num_panels = 2
+        showing_predictions = "prediction" in sample
+        if showing_predictions:
+            predictions = sample["prediction"].numpy()
+            num_panels += 1
+        
+        # converting 5 classes to [0,1,2,3,4] for the plot function.
```

Comment:
We can do this when we define cmap. Honestly, instead of using cmap, why not remove it and divide the mask/prediction by 4 so the image is in the 0–1 range? Then we don't need to do any fancy cmap stuff, we just plot the raw grayscale values.